src/mmrelay/meshtastic_utils.py

```
@@ -3303,6 +3351,149 @@ def _get_connection_retry_wait_time(attempts: int) -> float:
+            "Skipping connect-time metadata probe; metadata probe already in progress"
+        )
+        return
+
+    logger.debug(
+        "Scheduled one-shot connect-time metadata probe (timeout=%.1fs)",
+        timeout_secs,
+    )
+
+
+def _rollback_connect_attempt_state(
+    client: Any,
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `client`

(ANN401)